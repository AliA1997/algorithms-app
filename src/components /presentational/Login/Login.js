import React from 'react';
import LoginPage from '../../container/LoginPage/LoginPage';

const Login = () => {
    return (
        <LoginPage>
            <div className='login-container'>
                <div className='login-wrapper'>
                    <label className='login-label'>Username</label>
                    <input type='text' onChange={props.onChange} value={props.username} className='login-input' />
                    <label className='login-label'>Password</label>
                    <input type='password' onChange={props.onChange2} value={props.password} className='login-input' />
                    <button className='login-btn'>Login</button>
                </div>
            </div>
        </LoginPage>    
    );
};

export default Login;