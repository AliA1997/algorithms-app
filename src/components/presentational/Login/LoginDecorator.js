//import withHandlers to create a hoc 
import { withHandlers } from 'recompose';

const decorator = withHandlers({
    onClick: ({login, history}) => e => {
        e.preventDefault();
        login();
        history.push('/');
    },
    onClick2: ({history}) => () => {
        history.push('/register');
    },
    handleChange1: ({handleUsername}) => (e) => {
        handleUsername(e.target.value)
    },
    handleChange2: ({handlePassword}) => (e) => {
        handlePassword(e.target.value);
    }
})


export default decorator;