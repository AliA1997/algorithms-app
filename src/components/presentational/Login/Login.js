import React from 'react';
//import css file for styling 
import './Login.css';
import { Link } from 'react-router-dom';

export const Login = (props) => {
    return (
        <div className='login-container'>
            <div className='login-wrapper'>
                <h1 className='login header'>Login</h1>
                <label className='form-label'>Username</label>
                <input type='text' onChange={props.onChangeUsername} value={props.username} className='login-input' />
                <label className='form-label'>Password</label>
                <input type='password' onChange={props.onChangePassword} value={props.password} className='login-input' />
                <label className='form-label'>Don't have a Account?</label>
                <button className='btn'>Login</button>
            </div>
        </div>
    );
};


export default Login;