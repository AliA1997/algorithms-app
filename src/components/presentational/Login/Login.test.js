//Make sure to import react 
import React from 'react';
//Sample Component
import { Login } from './Login';
//import the renderer from react-test-render to create a mock component.
import render from 'react-test-renderer';

//Now test if the componet matches the snapshot.
describe('Does it match the snapshot', () => {
    //Test if it matches the snapshot
    test('Test if it matches snapshot', () => {
        //Create mock componet and convert it to json so it can check if it matches the snapshot, since the snapshot is json.
        const tree = render.create(<Login />).toJSON();
        //Now match the tree
        expect(tree).toMatchSnapshot(tree);
    })
})