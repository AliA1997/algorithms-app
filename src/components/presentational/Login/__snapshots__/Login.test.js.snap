// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
<div
  className="login-container"
>
  <div
    className="login-wrapper"
  >
    <h1
      className="login header"
    >
      Login
    </h1>
    <label
      className="form-label"
    >
      Username
    </label>
    <input
      className="login-input"
      onChange={undefined}
      type="text"
      value={undefined}
    />
    <label
      className="form-label"
    >
      Password
    </label>
    <input
      className="login-input"
      onChange={undefined}
      type="password"
      value={undefined}
    />
    <label
      className="form-label"
    >
      Don't have a Account?
    </label>
    <button
      className="btn"
    >
      Login
    </button>
  </div>
</div>
`;
